index.js

```diff
@@ -3,191 +3,147 @@ const express=require('express');
-        }
-    },
-];
-
+mongoose.connect('mongodb://localhost:27017/myFlix', { useNewUrlParser: true, useUnifiedTopology: true });
 app.use(morgan('common'));
 
 //return list of all movies to client
 app.get('/movies',(req,res)=>{
-    res.json(movies);
+  Movies.find()
+  .then(movies=>res.json(movies))
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error: ' + error)
+  })
 });
 
-//return data about a single movie
+//return data about a single movie by title
 app.get('/movies/:title', (req, res) => {
-    res.json(movies.find((movie) =>movie.title === req.params.title));
-  });
-
-//return data about a genre by genre name
-app.get('/movies/genre/:name', (req, res) => {
-    const movieGenre = movies.find(movie => movie.genre.name === req.params.name);
-    if (movieGenre) {
-        res.status(200).json(movieGenre.genre.description);
-    } else {
-        res.status(400).send(req.params.name + ' not found')
-    }
+  Movies.findOne({Title:req.params.title})
+  .then(movie=>res.json(movie))
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error: ' + error)
+  })
 });
 
-//return data about a director by name
-app.get('/movies/director/:name',(req,res)=>{
-    const movieDirector=movies.find(movie=>movie.director.name=req.params.name);
-    if(movieDirector){
-      res.status(200).json(movieDirector.director)
-    }else{
-      res.status(400).send('No movie directed by' + req.params.name + ' found')
-    }
-  });
+//return description of a genre by genre name
+app.get('/movies/genre/:genreName', (req, res) => {
+  Movies.findOne({"Genre.Name":req.params.genreName})
+  .then(movie=>res.json(movie.Genre.Description))
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error: ' + error)
+  })
+});
 
-//get all users
-app.get('/users',(req,res)=>{
-    res.json(users);
-  });
-  
-//Allow  new users to register
-  app.post('/users',(req,res)=>{
-    let newUser=req.body;
-    if(!newUser.username){
-      res.status(400).send('Username is missing in the request body')
-    }else{
-      newUser.id=uuid.v4();
-      users.push(newUser);
-      res.status(201).send(newUser)
-    }
-  });
+//return data about a director by director's name
+app.get('/movies/director/:directorName',(req,res)=>{
+  Movies.findOne({"Director.Name":req.params.directorName})
+  .then(movie=>res.json(movie.Director))
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error: ' + error)
+  })
+});
 
-//Allow users update their  password 
-app.put('/users/:username/:password', (req,res)=>{
-    let user=users.find(user=>user.username===req.params.username);
+//Create user
+app.post('/users',(req,res)=>{
+  Users.findOne({Username:req.body.Username})
+  .then(user=>{
     if(user){
-      user.password=req.params.password
-      res.status(201).send( user.username +'\'s passowrd has been succesfully updated')
+      res.status(400).send(req.body.Username + ' already exists!')
```

Comment:
Always keep the naming consistent, `Username` and `username`  keep one across the codebase